Gui/opensim/CMakeLists.txt

```diff
@@ -16,6 +16,35 @@ endif()
 
 find_package(Ant REQUIRED)
 
+# http://xit0.org/2013/04/cmake-use-git-branch-and-commit-details-in-project/
+# Get the latest abbreviated commit hash of the working branch
+find_package(Git)
+if(NOT Git_FOUND)
+  set(GIT_COMMIT_HASH "nocommit")
+  set (VERSION "000")
+else()
+    execute_process(
+      COMMAND git describe --always
+      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+      OUTPUT_VARIABLE GIT_COMMIT_HASH
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+	# prepend date so that build ids can meaningfully be sorted
+    execute_process(
+      COMMAND git show -s --format=%cd --date=short HEAD
+      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+      OUTPUT_VARIABLE GIT_COMMIT_DATE
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+    set (VERSION "${GIT_COMMIT_DATE}-${GIT_COMMIT_HASH}")
+endif()
+# inform user of created build id
+message(STATUS "Build VERSION: ${VERSION}")
```

Comment:
Ideally we would have additional logic here so that if the current commit has an annotated tag, then the version is simply the tag name, with no commit hash or date. That way, the official 4.0 release will be named `OpenSim 4.0` without having to do anything special.

Reply:
I'm fine with creating a dummy tag. We can also test with a dummy repo.